backends/simplego/dotgeneral.go

```diff
@@ -223,38 +277,85 @@ func execDotGeneral(backend *Backend, node *Node, inputs []*Buffer, _ []bool) (*
 	output := backend.getBufferForShape(outputShape)
 	output.Zeros()
 
-	// Problem size (per example of the batch):
+	// Check if inputs were pre-blocked at graph build time (via BlockForDotGeneral node).
+	lhsNode := node.inputs[0]
+	rhsNode := node.inputs[1]
+	lhsBlocked := lhsNode.opType == backends.OpTypeBlockForDotGeneral
+	rhsBlocked := rhsNode.opType == backends.OpTypeBlockForDotGeneral
+
+	// Handle pre-blocked cases
+	switch {
+	case lhsBlocked && rhsBlocked:
+		// Both inputs are pre-blocked - most efficient path
+		lhsBlockData := lhsNode.data.(*blockForDotGeneralData)
+		rhsBlockData := rhsNode.data.(*blockForDotGeneralData)
+		if err := execDotGeneralWithBothBlocked(backend, lhs, rhs, lhsBlockData, rhsBlockData, params, output); err != nil {
+			backend.putBuffer(output)
+			return nil, err
+		}
+		return output, nil
+
+	case lhsBlocked:
```

Comment:
Do we need to handle the cases with only one side blocked ? 

I suspect we can either block both sizes or none ?
My impression is that if one side is blocked the extra cost of blocking the other is  not much extra cost...

Reply:
I unified these into a single `execDotGeneralBlockedUnified` function for now 

When only one-side is blocked, I made a change to use the blocked path for both sides (ie the non-blocked side gets blocked at runtime

I ran some benchmark tests and found that pre-blocking provides a 3-3.5x speedup even with small LHS's. While the main speedup comes from blocking the RHS (weights), it seems like the overhead of blocking the LHS is not much extra cost and still out-performs the baseline.